curriculum/challenges/english/07-scientific-computing-with-python/learn-special-methods-by-building-a-vector-space/65f055b9190fc41ca35549b8.md

```diff
@@ -0,0 +1,38 @@
+---
+id: 65f055b9190fc41ca35549b8
+title: Step 3
+challengeType: 20
+dashedName: step-3
+---
+
+# --description--
+
+Python provides a variety of methods having both a heading and a trailing double underscore in their name. You have already met some, such as `__init__` and `__str__`. These methods having the name in the form of `__<name>__` are known as special methods.
+
+They are called under the hood in specific situations (e.g. `__init__` during instantiation, `__str__` when the object is printed), affecting the behaviour of the class which defines them. Going on with this project, you are going to learn some of the most commonly used special methods.
```

Comment:
It might be convoluted, but I'm not saying that calling them affects the class behaviour
>affecting the behaviour of the class which defines them

I'll think about how to reword it.